thirdeye/thirdeye-frontend/app/pods/components/entity-filter/component.js

```diff
@@ -0,0 +1,48 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  init() {
+    this._super(...arguments);
+    const filterBlocks = this.get('filterBlocks');
+
+    // Set up filter block object
+    filterBlocks.forEach((block) => {
+      block.filtersArray = [];
+      block.isHidden = false;
+      // Dedupe and remove null or empty values
+      block.filterKeys = Array.from(new Set(block.filterKeys.filter(value => Ember.isPresent(value))));
+      block.filterKeys.forEach((filter) => {
+        block.filtersArray.push({
+          isActive: false,
+          name: filter,
+          id: filter.dasherize()
+        });
+      });
+    });
+  },
+
+  alertFilters: [],
+
+  actions: {
+    /**
+     * Handles selection of filter items. Each time a filter is selected, this component will
+     * pass an array of selected filters to its parent.
+     */
+    onFilterSelection(category, filterObj) {
+      Ember.set(filterObj, 'isActive', !filterObj.isActive);
```

Comment:
Looks like a good candidate for `this.toggleProperty`

Reply:
After some investigation, I found that `toggleProperty` will not be able to find 'filterObj', as it is not an observed property.